ibis-server/tests/routers/v2/connector/test_snowflake.py

```diff
@@ -261,14 +261,45 @@ def test_validate_rule_column_is_valid_without_one_parameter(manifest_str):
         assert response.status_code == 422
         assert response.text == "Missing required parameter: `modelName`"
 
-    @pytest.mark.skip(reason="Not implemented")
     def test_metadata_list_tables():
-        pass
+        response = client.post(
+            url=f"{base_url}/metadata/tables",
+            json={"connectionInfo": connection_info},
+        )
+        assert response.status_code == 200
+        tables = response.json()
+        assert len(tables) == 8
+        table = next(filter(lambda t: t["name"] == "TPCH_SF1.ORDERS", tables))
+        assert table["name"] == "TPCH_SF1.ORDERS"
+        assert table["primaryKey"] is not None
+        assert table["description"] == "Orders data as defined by TPC-H"
+        assert table["properties"] == {
+            "catalog": "SNOWFLAKE_SAMPLE_DATA",
+            "schema": "TPCH_SF1",
+            "table": "ORDERS",
+        }
+        assert len(table["columns"]) == 9
+        column = next(filter(lambda c: c["name"] == "O_COMMENT", table["columns"]))
+        assert column == {
+            "name": "O_COMMENT",
+            "nestedColumns": None,
+            "type": "TEXT",
+            "notNull": True,
+            "description": None,
+            "properties": None,
+        }
 
-    @pytest.mark.skip(reason="Not implemented")
     def test_metadata_list_constraints():
-        pass
+        response = client.post(
+            url=f"{base_url}/metadata/constraints",
+            json={"connectionInfo": connection_info},
+        )
+        assert response.status_code == 200
```

Comment:
Hi @ongdisheng,
Could you add expected like `test_metadata_list_tables`?
You did a good job. Thanks for your help.

Reply:
Sure, no problem.

Reply:
Hi @grieve54706, I've noticed that the ```TPCH_SF1``` schema doesn’t seem to enforce foreign key constraints, which causes the ```SHOW IMPORTED KEYS``` command to return an empty list.

Here are the results I found when querying the ```TABLE_CONSTRAINTS```:
![image](https://github.com/user-attachments/assets/41049ee9-9b2d-487a-ae79-7b70e98f34a5)

Given this, would it be appropriate to add an assertion to check that the result returns an empty list? I’d be glad to hear any other suggestions you may have. Thank you!

Reply:
Yes, asserting it is an empty list is a good choice.